packages/core/versioning/core-version-http-server/src/version_http_toolkit.ts

```diff
@@ -154,12 +106,29 @@ export interface VersionedRouter<Ctx extends RqCtx = RqCtx> {
   options: VersionedRouteRegistrar<'options', Ctx>;
 }
 
+/** @experimental */
+export type InputValidation<P, Q, B> = RouteValidatorFullConfig<P, Q, B>;
+
+/** @experimental */
+export type OutputValidation<R> = RouteValidationFunction<R> | Type<R>;
+
+/**
+ * Versioned route validation
+ * @experimental
+ */
+interface FullValidationConfig<P, Q, B, R> {
+  /** Validation to run against route inputs: params, query and body */
+  in?: InputValidation<P, Q, B>;
+  /** Validation to run against route output */
+  out?: OutputValidation<R>;
```

Comment:
what about using `req` and `res` as opposed to `in`/`out`? Because we're not really trying to abstract away that this is "raw" http this feels a bit more intuitive for a consumer.

I also wonder if `out` / `res` should be a "namespace" to be able to support other validation like the headers. But can't really think of a good reason we'd need that.

Reply:
Yeah, I like both of those ideas! Namespacing `res` is a bit more defensive and doesn't hurt readability.

I went with `request` and `response` as I felt this is a bit more consistent with the rest of the route config naming.